### DISCUSSION TITLE
Question regarding alarm wakeup "hours" setting

### DISCUSSION THREAD

**Discussion by gobees** [PiSupply/PiJuice#975] [Q&A]
Regarding the hours setting, what happens when the UTC hour toggles over to the last hour in the hours setting?   For example, if I have hours set for:  "21;22;23" , what will the wakeup alarm do between the hours of 23 and 24?  My minutes setting is set at 20 minute period.   I would assume that the pi would start three times with these hour and minutes period settings but my testing did not reveal that and that could be something I did wrong during the testing.

I just want to know if for that last hour, do I need to add the next hour so that the entire hour of 23 (in my example) will be used. So, assuming I want three wakeups after 23:00 but before 24:00, do I need to set my hours setting to "0;21;22;23" ?  or "21;22;23;24"?

Hope this makes sense.

thanks.

**Comment by quitmeyer** [comment]
hi, i don't think there is a "24" hour you can even put in, it would be 0

Also i don't totally understand your questions and what you mean by "the entire hour will be used?"

if you have 
hours: 21;22;23
minutes;20

then 3 wakeup calls will be issued, one at 21:20, 22:20, and 23:20

**Reply by gobees** [reply]
you say "0" for midnight, I say "24", tomato, tomaaa-to.

I'm saying with my testing, the wake up alarm did not occur as I had predicted based on settings.  You seem to be saying that it does occur (once each hour in your example) and that was not my experience.

Also note that minute "period" is what I'm using to set the alarm, not "minutes".  so with a 20 minute "period" set, I expect the alarm to occur three times each hour.

**Reply by quitmeyer** [reply]
ahh ok i understand, i didn't realize you had minutes period. yeah it should go 2 or three times each hour. im not sure if it will count 00 as a minute to go, but i would think it would.

i had probs with it not running because the wakeup alarm wouldn't re-set itself, but if you look in the discussion of this github, i just shared some code to reschedule everything everytime it starts up, and it's been working good for me :)

**Reply by gobees** [reply]
Yeah.   Very early on, I took the same approach and left nothing to chance -- meaning I included code that reset the alarm after each start of the pi.   I'm now almost a year into this project and I'm finding the pijuice to be somewhat unpredictable.  I have two different pi's, and two pijuice's attached and they both hiccup on me from time to time.
